### ISSUE TITLE
Option: `setReturnFocus` is not being respected on component unmount

### ISSUE THREAD

**Issue by lukwallace** [focus-trap/focus-trap-react#89]
`setReturnFocus` seems to be ignored in the `focusTrapOptions`; went looking for solutions in the original `focus-trap` repository, then realized the deactivation logic is being hijacked for the react variant. It doesn't look like it was taken into account during implementation.

Going to use the more generic `onDeactivate` as a workaround for the time being; would love to see an update to README though, in case anyone else stumbles into this issue as well.

**Comment by lukwallace** [comment]
Ack -- strike that; you cannot use `onDeactivate` to define focus b/c it is invoked before the hijacking 😞 
```
function componentWillUnmount() {
      this.focusTrap.deactivate();

      if (this.props.focusTrapOptions.returnFocusOnDeactivate !== false && this.previouslyFocusedElement && this.previouslyFocusedElement.focus) {
        this.previouslyFocusedElement.focus();
      }
    }
```
I will try even hacky-er solution -- change focus to element I actually want to receive focus before activating trap...

**Comment by liamsc** [comment]
@lukwallace I was able to get this working by calling an onClose method as part of my onDeactivate function:
```
            <FocusTrap focusTrapOptions={{ onDeactivate: this.onDeactivate, initialFocus: this.setInitFocus }}>
```

```
    onDeactivate(){
        this.props.closeModal();
        if (this.props.onClose) {
            this.props.onClose();
        }
    }
```

in the props.onClose method I am calling `openModalButton.current.focus()` where openModalButton is a ref to the button element. Hope this helps! I was having a similar issue where the setReturnFocus function didn't appear to get called.

**Comment by stefcameron** [comment]
@lukwallace I'm a new maintainer of all 3 related repos, and slowly catching-up here. I think the code that was causing you issue has changed significantly since you first posted this issue. The code no longer has any reference to the `setReturnFocus` option that I can see. The code "hijacks" the `returnFocusOnDeactivate` flag from `focus-trap` (for reasons documented in code comments), but still attempts to perform the same function, so there shouldn't be any need to update the README.

@liamsc Thanks for posting your solution in case it helps @lukwallace and others.

Closing. Please LMK if the issue still isn't fixed. If so, I'd like to understand more about the specific use case causing the problem.

**Comment by Dan503** [comment]
This was fixed in PR #394
